src/fiber/graph.rs

```diff
@@ -748,31 +782,27 @@ where
         }
 
         let mut current = source_node.node_id;
-        loop {
-            if let Some(elem) = distances.get(&current) {
-                let next_hop = elem.next_hop.as_ref().expect("next_hop is none");
-                result.push(PathEdge {
-                    target: next_hop.0,
-                    channel_outpoint: next_hop.1.clone(),
-                });
-                current = next_hop.0;
-            } else {
-                break;
-            }
+        while let Some(elem) = distances.get(&current) {
+            let next_hop = elem.next_hop.as_ref().expect("next_hop is none");
+            result.push(PathEdge {
+                target: next_hop.0,
+                channel_outpoint: next_hop.1.clone(),
+            });
+            current = next_hop.0;
```

Comment:
suggest to use remove:
```suggestion
        while let Some(elem) = distances.remove(&current) {
            let (next_pubkey, next_out_point) = elem.next_hop.expect("next_hop is none");
            result.push(PathEdge {
                target: next_pubkey,
                channel_outpoint: next_out_point,
            });
            current = next_pubkey;
            if current == target {
                break;
            }
        }
```